File: benchmark/src/lua/floyd_warshall.lua
local n=150;local INF=999999999;local dist={}
for i=1,n do dist[i]={};for j=1,n do dist[i][j]=i==j and 0 or INF end end
local seed=12345;local function ri(m)seed=(seed*16807)%2147483647;return seed%m end
for i=1,n do for j=1,4 do local t=(i-1)*7+(j-1)*13+1;t=t%n+1;if t~=i then dist[i][t]=1+((i-1+j-1)%50)end end end
local start=os.clock()
for k=1,n do for i=1,n do local dik=dist[i][k];if dik~=INF then for j=1,n do local nd=dik+dist[k][j];if nd<dist[i][j]then dist[i][j]=nd end end end end end
local reach,sm=0,0;for i=1,n do for j=1,n do if dist[i][j]<INF then reach=reach+1;sm=sm+dist[i][j]end end end
local finish=os.clock();print("Result: "..reach.."_"..(sm%1000003));print(string.format("Time: %.0fms",(finish-start)*1000))

File: benchmark/src/perl/floyd_warshall.pl
use Time::HiRes qw(time);
my$n=150;my$INF=999999999;my@dist;for my$i(0..$n-1){for my$j(0..$n-1){$dist[$i][$j]=$i==$j?0:$INF}}
my$seed=12345;my$ri=sub{my$m=shift;$seed=($seed*16807)%2147483647;$seed%$m};
for my$i(0..$n-1){for my$j(0..3){my$t=($i*7+$j*13+1)%$n;$dist[$i][$t]=1+(($i+$j)%50)if$t!=$i}}
my$start=time();
for my$k(0..$n-1){for my$i(0..$n-1){my$dik=$dist[$i][$k];if($dik!=$INF){for my$j(0..$n-1){my$nd=$dik+$dist[$k][$j];$dist[$i][$j]=$nd if$nd<$dist[$i][$j]}}}}
my($reach,$sm)=(0,0);for my$i(0..$n-1){for my$j(0..$n-1){if($dist[$i][$j]<$INF){$reach++;$sm+=$dist[$i][$j]}}}
my$end=time();my$ms=($end-$start)*1000;print"Result: ${reach}_".($sm%1000003)."\n";printf"Time: %.0fms\n",$ms;
